lib/licensed/sources/gradle.rb

```diff
@@ -9,127 +9,156 @@ module Licensed
+        JSON.parse(gradle_command("printDependencies", executable: gradle_executable, configurations: configurations)).map do |package|
           name = "#{package["group"]}:#{package["name"]}"
           Dependency.new(
             name: name,
             version: package["version"],
-            path: File.join(@config.pwd, GRADLE_LICENSES_PATH, name),
+            path: gradle_licenses_path,
+            executable: gradle_executable,
+            configurations: configurations,
             metadata: {
               "type"     => Gradle.type
             }
           )
         end
       end
 
-      def with_latest_licenses
-        download_and_cache_licenses do
-          yield
-        end
-      end
-
       private
 
-      def download_licenses_from_csv
-        path = File.join(@config.pwd, GRADLE_LICENSES_PATH)
-        downloaded_licenses = {}
-        CSV.foreach(File.join(path, GRADLE_LICENSES_CSV_NAME), headers: true) do |row|
-          url = row["moduleLicenseUrl"]
-          artifact, _, version = row["artifact"].rpartition(":")
-          file_path = File.join(path, artifact, "LICENSE")
-          FileUtils.mkdir_p(File.join(path, artifact))
-          if cache_path = downloaded_licenses[url]
-            FileUtils.cp(cache_path, file_path)
-          else
-            uri = URI(row["moduleLicenseUrl"])
-            File.write(file_path, Net::HTTP.get(uri))
-            downloaded_licenses[url] = file_path
-          end
-        end
+      def gradle_licenses_path
+        @gradle_licenses_path ||= File.join(config.pwd, GRADLE_LICENSES_PATH)
       end
 
-      def download_and_cache_licenses
-        gradle_command("generateLicenseReport")
-        download_licenses_from_csv
-        yield
-        FileUtils.rm_rf(File.join(@config.pwd, GRADLE_LICENSES_PATH))
+      def gradle_executable
+        return @gradle_executable if defined?(@gradle_executable)
+        @gradle_executable = begin
+          gradlew = File.join(config.pwd, "gradlew")
+          return gradlew if File.executable?(gradlew)
+          return "gradle" if Licensed::Shell.tool_available?("gradle")
+        end
       end
 
-      # Returns the output from running `npm list` to get package metadata
-      def package_metadata_command
-        gradle_command("printDependencies")
+      # Returns the configurations to include in license generation.
+      # Defaults to ["runtime", "runtimeClasspath"]
+      def configurations
+        @configurations ||= begin
+          if configurations = config.dig("gradle", "configurations")
+            Array(configurations)
+          else
+            DEFAULT_CONFIGURATIONS
+          end
+        end
       end
 
-      def gradle_command(*args)
-        Tempfile.open(["license", ".gradle"], @config.pwd) do |f|
-          f.write gradle_file
+      def self.gradle_command(*args, executable:, configurations:)
+        Tempfile.create do |f|
```

Comment:
I made a (possibly wrong) assumption that the location of the temporary file doesn't matter, and the outputs of the gradle command would be in the local directory.  If that's not the case, then the gradle command needs to be given a path that would be used instead of replace `@config.pwd`.

Also not sure why I changed open to create, probably just miming something I read from the web.  The end result should be the same I think.